docs/proposals/2841-flux-hpc/README.md

```diff
@@ -0,0 +1,185 @@
+
+We will add the `Flux` field to the `MLPolicySource` struct, which is embedded within the `MLPolicy`.
+
+```go
+type MLPolicySource struct {
+    Torch *TorchMLPolicySource `json:"torch,omitempty"`
+    MPI   *MPIMLPolicySource   `json:"mpi,omitempty"`
+
+    // FluxMLPolicy defines policy only for Flux
+    // +optional
+    Flux  *FluxMLPolicySource  `json:"flux,omitempty"`
+}
+```
+
+For the `FluxMLPolicySource`, we define the minimum required parameters needed for Flux and installing the view. The view must be compatible with the application container.
+
+```go
+// FluxMLPolicySource represents a Flux HPC runtime configuration.
+type FluxMLPolicySource struct {
+
+    // numNodes is the number of physical nodes for the job.
+    // This is defined a level up on the Trainer
+
+    // numProcPerNode is the number of processes per node.
+    // Defaults to 1.
+    // +kubebuilder:default=1
+    // +optional
+    NumProcPerNode *int32 `json:"numProcPerNode,omitempty"`
+
+    // Container image to use for Flux view that installs Flux
+    // This must be compatible with the application container
+    // Get the flux view container (these are choices)
+    // ghcr.io/converged-computing/flux-view-rocky:arm-9
+    // ghcr.io/converged-computing/flux-view-rocky:arn-8
+    // ghcr.io/converged-computing/flux-view-rocky:tag-9
+    // ghcr.io/converged-computing/flux-view-rocky:tag-8
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-noble
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-jammy
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-focal
+    // ghcr.io/converged-computing/flux-view-ubuntu:arm-jammy
+    // ghcr.io/converged-computing/flux-view-ubuntu:arm-focal
+    // We use an ubuntu (more recent) default since it is common
+    // +kubebuilder:default="ghcr.io/converged-computing/flux-view-ubuntu:arm-jammy"
+    Image string `json:"image,omitempty"`
+
+    // Network device for flux to use
+    // +kubebuilder:default="eth0"
+    NetworkDevice string `json:"networkDevice,omitempty"`
+
+    // Queue policy for Flux to use
+    // +kubebuilder:default="fcfs"
+    QueuePolicy string `json:"queuePolicy,omitempty"`
+}
+```
+
+Note that we are not exposing interactive or other options that are widely used, but not required.
+
+**Example `ClusterTrainingRuntime`:**
+
+This is an example created by an administrator.
+
+```yaml
+apiVersion: trainer.kubeflow.org/v1alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: flux-runtime
+  labels:
+    trainer.kubeflow.org/framework: flux
+spec:
+  mlPolicy:
+    flux:
+      numNodes: 1
+  template:
+    spec:
+      replicatedJobs:
+        - name: trainer
+          template:
+            spec:
+              completionMode: Indexed
+              completions: 1
+              parallelism: 1
+              template:
+                spec:
+                  containers:
+                    - name: node
+                      image: "placeholder-image"
```

Comment:
Will we run `flux run` directly in this replicatedJob after initContainer is finished?
I am asking whether we need to have two separate rJob, like in MPI: https://github.com/kubeflow/trainer/blob/master/manifests/base/runtimes/deepspeed_distributed.yaml#L24